src/app/(BottomTabNavigation)/Cases/QRCodeScanner/AddCase/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { useLocalSearchParams, useRouter } from 'expo-router';
+import React, { useEffect, useState } from 'react';
+import { Image, Text, TouchableOpacity, View } from 'react-native';
+import { getCase, uploadCase } from '../../../../../supabase/queries/case';
+import styles from './styles';
+
+enum caseStatus {
+  NotStarted,
+  InProgress,
+  Settled,
+  Cancelled,
+}
+
+export interface Case {
+  id: number;
+  approved: boolean;
+  title: string;
+  summary: string;
+  image: string;
+  caseStatus: caseStatus;
+}
+
+function AddCase() {
+  const local = useLocalSearchParams();
+  const { caseId } = local;
+  const router = useRouter();
+
+  const [dataImage, setDataImage] = useState('');
+  const [dataTitle, setDataTitle] = useState('');
+  const [dataSummary, setDataSummary] = useState('');
+
+  const parseData = (data: any) => {
+    const myData = {
+      ...(data[0] as Partial<Case>),
+    };
+    return myData as Case;
+  };
+
+  useEffect(() => {
+    const getData = async () => {
+      const { data, error } = await getCase(caseId);
+      if (error) {
+        console.log('error');
+      }
+
+      const myData = parseData(data);
+      const { image, title, summary } = myData;
+
+      setDataImage(image);
+      setDataTitle(title);
+      setDataSummary(summary);
+      console.log(data);
+    };
+    getData();
+  }, []);
+
+  const addToCases = async () => {
+    uploadCase(caseId);
+  };
+  return (
+    <View style={styles.container}>
+      <Text style={styles.title}>{dataTitle}</Text>
```

Comment:
Now, instead of using state variables, which have a bit of a delay, we can use the parameters we pass down from the QR Code Scanner screen!